text/0059-otlp-trace-data-format.md

```diff
@@ -0,0 +1,348 @@
+# OTLP Trace Data Format
+
+_Author: Tigran Najaryan, Splunk_
+
+**Status:** `approved`
+
+OTLP Trace Data Format specification describes the structure of the trace data that is transported by OpenTelemetry Protocol (RFC0035).
+
+## Motivation
+
+This document is a continuation of OpenTelemetry Protocol RFC0035 and is necessary part of OTLP specification.
+
+## Explanation
+
+OTLP Trace Data Format is primarily inherited from OpenCensus protocol. Several changes are introduced with the goal of more efficient serialization. Notable differences from OpenCensus protocol are:
+
+1. Removed `Node` as a concept.
+2. Extended `Resource` to better describe the source of the telemetry data.
+3. Replaced attribute maps by lists of key/value pairs.
+4. Eliminated unnecessary additional nesting in various values.
+
+Changes 1-2 are conceptual, changes 3-4 improve performance.
+
+## Internal details
+
+This section specifies data format in Protocol Buffers.
+
+### Resource
+
+```protobuf
+// Resource information. This describes the source of telemetry data.
+message Resource {
+  // labels is a collection of attributes that describe the resource. See OpenTelemetry
+  // specification semantic conventions for standardized label names:
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md
+  repeated AttributeKeyValue labels = 1;
```

Comment:
Why do we need to introduce the term "labels"? It feels confusing and adding little value over "attributes".

Reply:
This is not new in this proposal. "labels" are the term used in the specification currently: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-resource.md

I agree it would likely make sense to make this uniform. I created an issue for that https://github.com/open-telemetry/opentelemetry-specification/issues/342

Reply:
Resolving, issue created to track this separately.